firestore/firestore_snippets/listen_changes.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+// [START fs_listen_changes]
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"cloud.google.com/go/firestore"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// listenChanges listens to a query, returning the list of document changes.
+func listenChanges(ctx context.Context, w io.Writer, projectID string) ([]firestore.DocumentChange, error) {
+	// projectID := "project-id"
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+
+	client, err := firestore.NewClient(ctx, projectID)
+	if err != nil {
+		return nil, fmt.Errorf("firestore.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	modified := false
+	changes := make([]firestore.DocumentChange, 0)
```

Comment:
If we need to return the changes (not sure we do):
```suggestion
	changes := []firestore.DocumentChange{}
```

Reply:
it's been removed